modules/code-builder/docs/PRD.md

```diff
@@ -0,0 +1,177 @@
+
+# Product Requirements Document (PRD): Masonry Framework
+
+## 1. Bricks
+
+### a. Brick Types
+
+#### 1. Data Bricks: These represent simple data values, such as numbers or strings, and serve as inputs for other bricks.
+   - Number Data Brick (representing note values, durations, etc.)
+
+     ![duration bricks](./images/image.png)
+   - String Data Brick (representing note names, pitches, etc.)
+
+     ![pitches and note names](./images/image-1.png)
+
+#### 2. Expression Bricks: These perform logical or mathematical operations.
+   - Equal Expression Brick
+   - Not Equal Expression Brick
+   - Greater Than Expression Brick
+   - Less Than Expression Brick
+   - And Expression Brick
+   - Or Expression Brick
+   - Not Expression Brick
+   - Random Expression Brick
+   - Length Expression Brick
+   - Letter Expression Brick
+   - Number Expression Brick
+
+     ![alt text](./images/image-2.png)
+
+#### 3. Statement Bricks: These define actions to be taken.
+   - Play Note Statement Brick
+   - Rest Statement Brick
+   - Set Instrument Statement Brick
+   - Increase Volume Statement Brick
+   - Decrease Volume Statement Brick
+   - Set Tempo Statement Brick
+   - Set Key Statement Brick
+   - Set Mode Statement Brick
+   - Start Drum Statement Brick
+   - Start Chord Statement Brick
+
+     ![example](./images/image-3.png)
+
+#### 4. Brick Bricks:
+   - Repeat Brick
+   - Forever Brick
+   - If Brick
+   - If-Else Brick
+   - Action Brick
+   - Start Brick
+   - Clamp Brick
+   - Store In Brick
+   - Box Brick
+   - Pitch Brick
+   - Rhythm Brick
+   - Meter Brick (representing time signatures)
+   - Volume Brick
+   - Articulation Brick
+   - Pitch-Time Matrix Brick
+
+### b. Brick Appearance
+   - **Distinct Shapes**: Each brick type has a unique shape to differentiate its function visually.
+   - **Colors**: Color-coding helps users quickly identify the type of brick.
+   - **Icons and Labels**: Visual symbols and text labels indicate the brick's function.
+   - **Input/Output Ports**: Connectors visually represent where bricks can attach to each other.
+   - **Editable Text Labels/Fields**: Users can input data directly into the bricks, such as note names, durations, and numerical values.
+
+### c. Brick Interactions
+   - **Inline Text Editing**: Users can click on text fields within bricks to edit labels and values directly.
+   ![alt text](./images/image-4.png)
+   - **Drop-Down Menus/Selectors**: For choosing predefined options, enhancing ease of use.
+   - **Dedicated Editors/Dialogs**: For more complex properties, a separate interface allows detailed configuration.
+   ![alt text](./images/image-5.png)
```

Comment:
I think pie menus could be their own section. I personally like it over dropdowns but what we currently have right now isn't that great of an experience either. Might want to discuss some shortcomings and potential improvements.